keps/693-multikueue/README.md

```diff
@@ -316,6 +323,72 @@ that ware created by this multikueue manager cluster and delete them if their lo
 - `WorkerLostTimeout` - defines the time a local workload's multikueue admission check state is kept Ready
 if the connection with its reserving worker cluster is lost.
 
+
+### MultiKueue Dispatcher API
+
+The MultiKueue Workload Controller follows a two-step process: first, it nominates worker clusters for each workload based on the configured dispatching mode;
+then, it synchronizes the workload by creating and managing its copies in the nominated clusters. 
+This ensures that only one cluster ultimately admits and runs the workload, while the management cluster maintains up-to-date status by tracking the admitted job.
+
+#### Cluster Nomination Phase
+Cluster nomination phase looks differently depending on the configured dispatching mode.
+MultiKueue supports three dispatching modes: AllAtOnce (`kueue.x-k8s.io/multikueue-dispatcher-all-at-once`),
+Incremental (`kueue.x-k8s.io/multikueue-dispatcher-incremental`), and External.
+
+* **AllAtOnce Mode**:  
+  In this mode, all available clusters are considered for dispatching at once, and the workload is copied to each worker cluster.
+
+* **Incremental Mode**:  
+  In this mode, clusters are nominated incrementally in rounds of fixed duration (5 minutes per round). 
+  The process begins by nominating an initial set of 3 clusters, which are added to the `.spec.multiKueueNominatedClusters` field.
+  During the first 5-minute round, only these clusters are eligible to admit the workload.
+  If none of the initial clusters admit the workload within the round's duration, the next round begins, and 3 additional clusters are nominated.
+  This incremental approach continues, adding 3 more clusters in each subsequent round, until the workload is admitted or all eligible clusters have been considered.
+  This strategy allows for a controlled and gradual expansion of candidate clusters, rather than dispatching the workload to all clusters at once,
+  which can help optimize resource usage and reduce unnecessary workload duplication.
+  The workload's status condition `HaveNominatedWorkers` stores the round duration starting time.
+  This allows the controller to determine when to advance to the next nomination round based on the elapsed time since the round began.
+
+* **External Mode**:  
+  In this mode, the selection of worker clusters is entirely delegated to an external controller. 
+  The external controller is responsible for setting the `.spec.multiKueueNominatedClusters` field to the names of the selected clusters.
+  If the nominated clusters field is changed by the external controller, workloads are removed from any clusters that are no longer nominated,
+  even if they were previously admitted there.
```

Comment:
I would suggest to make it immutable once the workload is admitted. There is a small race condition if someone wants to modify the multiKueueNominatedClusters, and we don't yet see the event that the workload is admitted. In this case, indeed, we may update the nominated clusters, but I would make it as a note, or even leave to implementation decisions. What is important is to make it immutable once we realize the workload is admitted.